### PR TITLE
coot/typed protocols

### PR DESCRIPTION
**Pull request by coot**
- Added typed-protocols-0.1.0.4
- Added typed-protocols-examples-0.2.0.0

### CODE REVIEW

**Comment by andreabedini** [comment]
This is annoying. typed-protocols needs a newer hackage but with a newer hackage something else breaks.

File: _sources/io-sim/0.3.0.0/meta.toml [hunk 1]
```diff
@@ -2,3 +2,7 @@ timestamp = 2022-10-26T00:00:00Z
 github = { repo = "input-output-hk/io-sim", rev = "0df93fe773e9b66ce35f3637128533e5b0029558" }
 subdir = 'io-sim'
 
+
+[[revisions]]
+number = 1
```
Comment (michaelpj) [inline _sources/io-sim/0.3.0.0/meta.toml:7]: you forgot to add this

**Comment by michaelpj** [comment]
@coot are you expecting the revision you just made to fix the Hackage issue, or should we investigate separately?

**Comment by coot** [comment]
Expecting is too strong, but yes I am exploring an idea how to fix it.  The only place I know where we depend on `deque` is in `io-sim-0.3`.

File: _sources/io-sim/0.3.0.0/meta.toml (hunk 1 above)
Comment (coot) [reply _sources/io-sim/0.3.0.0/meta.toml:7]: I think I messed up the commit, let me re-push.

**Comment by coot** [comment]
That wasn't it, I  have no other clue what's wrong.

**Comment by michaelpj** [comment]
Remote store building doesn't seem to give us the dependency trace, which is annoying (I raised https://github.com/nixbuild/feedback/issues/31). Building locally with `--builders` instead, I got the trace:
```
error: build of '/nix/store/83d9sr41jylfm2pcjbfww3nfr7dca8nq-deque-lib-deque-0.3.drv' on 'ssh-ng://eu.nixbuild.net' failed: builder for '/nix/store/83d9sr41jylfm2pcjbfww3nfr7dca8nq-deque-lib-deque-0.3.drv' failed with exit code 1
data-default-class-lib-data-default-class> building
error: builder for '/nix/store/83d9sr41jylfm2pcjbfww3nfr7dca8nq-deque-lib-deque-0.3.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/kmrj9nz91xsmk8x5b8a7yqlj3ksqmfj9-io-classes-lib-io-classes-0.3.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/05n1cpqbyyb1jnnc6vq54wprzwkb6v1c-ouroboros-network-api-lib-ouroboros-network-api-0.2.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ggb8496z8i3jcv39nl9sc6n9bqxfg68v-ouroboros-network-lib-ouroboros-network-0.5.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wl11i556m4qad595wq1x4jxg0cplilff-typed-protocols-lib-typed-protocols-0.1.0.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0q9f8h8656323nl3z9gxpwwmazskkqsp-ouroboros-network-0.5.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/al5zy9f76lz1nvxha1dfbx87zh1b34am-all-smoke-test-packages.drv' failed to build
```
and indeed it looks like `io-classes-0.3.0.0` depends on `deque`: https://github.com/input-output-hk/io-sim/blob/0df93fe773e9b66ce35f3637128533e5b0029558/io-classes/io-classes.cabal#L67

I've pushed the revision. It's weird: I wouldn't expect us to be able to fix things by just adding lower bounds. Why would cabal not pick the later version anyway if that works (which it apparently does!) and there was nothing constraining it to pick the older one?

**Comment by coot** [comment]
Thanks @michaelpj I should be able to take it from here.

**Comment by coot** [comment]
Yes, I am looking into it.

File: _sources/quickcheck-contractmodel/0.1.3.0/revisions/1.cabal [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      bytestring ^>= 0.10.12,
+      containers ^>= 0.6.5.1,
+      lens ^>= 5.0 || ^>= 5.1 || ^>= 5.2,
+      mmorph ^>= 1.2,
+      mtl ^>= 2.2.2,
+      pretty ^>= 1.1.3.6,
+      time ^>= 1.9.3,
+    -- QuickCheck dependencies
+      QuickCheck ^>= 2.14,
+      quickcheck-dynamic >= 3.0.2,
```
Comment (michaelpj) [inline _sources/quickcheck-contractmodel/0.1.3.0/revisions/1.cabal:91]:
```suggestion
      quickcheck-dynamic ^>= 3.0.2,
```
did you miss the edit?

Comment (coot) [reply _sources/quickcheck-contractmodel/0.1.3.0/revisions/1.cabal]: Yes, already pushed it.

Comment (coot) [reply _sources/quickcheck-contractmodel/0.1.3.0/revisions/1.cabal]: That was it. :tada:

Comment (michaelpj) [reply _sources/quickcheck-contractmodel/0.1.3.0/revisions/1.cabal]: Checked and it's already upstream, so no need to do that.

**Pull request merged**